uint8 id1
uint8 id2
uint8 id3
uint8 id4
int16 curr1
int16 curr2
int16 curr3
int16 curr4